File: hw_24_04/app/src/main/java/com/example/hw_24_04/MainActivity.java
package com.example.hw_24_04;

import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    ListView listView;
    List<String> list;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        listView = findViewById(R.id.listView);
        list = new ArrayList<>();

        FirebaseFirestore db = FirebaseFirestore.getInstance();

        db.collection("students")
                .get()
                .addOnSuccessListener(queryDocumentSnapshots -> {

                    for (DocumentSnapshot doc : queryDocumentSnapshots) {
                        String name = doc.getString("name");
                        Long age = doc.getLong("age");

                        list.add(name + " - " + age);
                    }

                    ArrayAdapter<String> adapter = new ArrayAdapter<>(
                            this,
                            android.R.layout.simple_list_item_1,
                            list
                    );

                    listView.setAdapter(adapter);
                });
    }
}

File: hw_03_04/app/src/main/java/com/example/hw_03_04/MainActivity.kt
package com.example.hw_03_04;

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val recyclerView: RecyclerView = findViewById(R.id.recyclerView)

        recyclerView.layoutManager = LinearLayoutManager(this)

        val chats = listOf(
            Chat("Jornal da Cidade Online", "https://www.jornaldacid...", "13:30", 1380),
            Chat("Hindustan Times", "India saw another big ju...", "08:50", 16),
            Chat("Financial Times", "Schools must always t...", "Wed", 19),
            Chat("Saved Messages", "Draft: Monospaced", "Wed", 97),
            Chat("Bitcoin Channel", "Happy Eid Mubarak ", "Fri", 32),
            Chat("xAI Grok", "Some interesting update...", "10:35", 5),
            Chat("News Ukraine", "Останні новини сьогодні", "Yesterday", 0)
        )

        val adapter = ChatAdapter(chats)
        recyclerView.adapter = adapter
    }
}